src/main/java/duke/commands/DeleteCommand.java

```diff
@@ -0,0 +1,43 @@
+package duke.commands;
+
+import duke.storage.Storage;
+import duke.task.InvalidTaskSelectedException;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.ui.Ui;
+import java.io.IOException;
+
+/**
+ * This class encapsulates the delete command.
+ */
+public class DeleteCommand extends Command {
+    private final int taskId;
+
+    /**
+     * Constructor of the DeleteCommand class.
+     *
+     * @param taskId The id of the task to be deleted.
+     */
+    public DeleteCommand(int taskId) {
+        this.taskId = taskId;
+    }
+
+    /**
+     * Executes the delete command by deleting the task in the in-memory list,
+     * and deleting the corresponding task on the txt file.
+     *
+     * @param tasks The current list of tasks.
+     * @param ui The UI of the Duke app.
+     * @param storage The storage manager for the Duke app.
+     * @throws InvalidTaskSelectedException Thrown exception when user inputs invalid task Id.
+     * @throws IOException Thrown IO exception when trying to delete tasks on storage.
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage)
```

Comment:
I like your javadoc descriptions, they are very clear and I can understand them easily the first time I read it 👍